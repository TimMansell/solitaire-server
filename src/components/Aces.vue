<template>
  <div class="aces">
    <div
      class="ace"
      v-for="(ace, index) in aces"
      :key="index">
      A
    </div>
  </div>
</template>

<script>
export default {
  name: 'Aces',
  data() {
    return {
      aces: 4,
    };
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.aces {
  display: flex;
  padding: 1rem;
  height: 100px;
}
.ace {
  border: 1px solid white;
  // height: 100px;
  margin-right: 1rem;
  width: 100px;
}
</style>
