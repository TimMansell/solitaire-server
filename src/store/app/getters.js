const getters = {
  isGameLoading: (state) => state.isGameLoading,
  isGameWon: (state) => state.isGameWon,
  isGameLost: (state) => state.isGameLost,
  isGamePaused: (state) => state.isGamePaused.isPaused,
  isGameActive: (state) => state.isGamePaused.isActive,
  isTimerPaused: (state) => state.isTimerPaused,
  isOverlayVisible: (state) => state.isOverlayVisible,
  timer: (state) => state.game.time,
  moves: (state) => state.game.moves,
  showRules: (state) => state.showRules,
  showNewGame: (state) => state.showNewGame,
  showHistory: (state) => state.showHistory,
  versionMatch: (state) => state.versionMatch,
};

export default getters;
