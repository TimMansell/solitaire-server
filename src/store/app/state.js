// Return state as function so we can reset it.
const state = () => ({
  isGameWon: false,
  isGameLost: false,
  isGamePaused: {
    isPaused: false,
    isActive: false,
  },
  isTimerPaused: false,
  game: {
    moves: 0,
    time: 0,
  },
  showRules: false,
  showNewGame: false,
  showHistory: false,
  versionMatch: true,
  showTableHelper: true,
});

export default state;
