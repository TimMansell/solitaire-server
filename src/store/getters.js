const getters = {
  shuffledCards: (state) => state.shuffledCards,
  boardCards: (state) => state.board.cards,
  rules: (state) => state.rules,
};

export default getters;
