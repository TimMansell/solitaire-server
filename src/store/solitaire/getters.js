const getters = {
  boardCards: (state) => state.board.cards,
  foundationCards: (state) => state.board.foundation,
  selectedCardId: (state) => state.selectedCardId,
  hasMoves: (state) => state.hasMoves,
  draggedCards: (state) => state.draggedCards,
  draggedCardsIDs: (state) => state.draggedCards.map(({ id }) => id),
};

export default getters;
