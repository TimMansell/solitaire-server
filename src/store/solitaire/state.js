import { columns } from '@/config/settings.json';

// Return state as function so we can reset it.
const state = () => ({
  cards: [[], [], [], [], [], [], [], []],
  foundation: [[], [], [], []],
  placeholders: columns,
  selectedCardId: null,
  hasMoves: true,
  draggedCards: [],
  isNewGame: true,
});

export default state;
