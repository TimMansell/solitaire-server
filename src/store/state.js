// Return state as function so we can reset it.
const state = () => ({
  board: {
    cards: [],
    foundation: [],
  },
  selectedCardId: null,
  isGameWon: false,
  hasMoves: true,
  totalGames: null,
  gameID: null,
});

export default state;
