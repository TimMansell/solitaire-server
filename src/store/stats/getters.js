const getters = {
  globalStats: (state) => state.globalStats,
  userStats: (state) => state.userStats,
  playerCount: (state) => state.playerCount,
  userGameCount: (state) => state.userGameCount,
  globalGameCount: (state) => state.globalGameCount,
  leaderboards: (state) => state.leaderboards,
  showStats: (state) => state.showStats,
  showLeaderboards: (state) => state.showLeaderboards,
};

export default getters;
