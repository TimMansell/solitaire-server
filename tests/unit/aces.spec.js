import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Aces from '@/components/Aces.vue';
import state from '@/store/state';
import getters from '@/store/getters';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Aces.vue', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state,
      getters,
    });
  });

  it('matches snapshot', () => {
    const wrapper = shallowMount(Aces, { store, localVue });

    expect(wrapper).toMatchSnapshot();
  });
});
