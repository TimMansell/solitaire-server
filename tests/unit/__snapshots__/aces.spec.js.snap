// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aces.vue matches snapshot 1`] = `
<div class="aces">
  <div class="ace">
    <!---->
  </div>
  <div class="ace">
    <!---->
  </div>
  <div class="ace">
    <!---->
  </div>
  <div class="ace">
    <!---->
  </div>
</div>
`;
